default filter expanded to true

weili-govtech opened the pull request:
**Changes**
- Update filter checkbox and filter item `expanded` prop to be true by default

**Changelog entry**
-   `filter-item` and `filter-checkbox` are now expanded by default

qroll commented:
code changes look fine but I don't have context on why the default is changed. for the PS use case, those filters are closed by default 🤔

qroll commented:
thanks @xyukianesa for sharing on the `expanded` bug in mobile view

this is an issue on DS side, I did not check if the item is mobile before applying the prop in [this change](https://github.com/LifeSG/react-design-system/pull/327). putting up a PR to fix this.

weili-govtech commented:
thanks for the DS fix, I have replaced the changes with a DS update and changed the expanded description in storybook to mention it is applicable to desktops only
